sphinx/ext/extlinks.py

```diff
@@ -108,6 +109,7 @@ def role(typ: str, rawtext: str, text: str, lineno: int,
             else:
                 title = caption % part
         pnode = nodes.reference(title, title, internal=False, refuri=full_url)
+        pnode["classes"].append(f"extlink-{html.escape(name)}")
```

Comment:
Technically speaking, the `name` will be used as a role name. In particular, it must be a simple name, namely it matches `'(?:(?!_)\w)+(?:[-._+:](?:(?!_)\w)+)*`. Therefore, I think you don't need to escape the name (and you will have a patch of exactly one line  + two lines of test, an improvement of 25%!)

Reply:
Nice, I've removed the `escape` 🚀